src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -1855,13 +1901,27 @@ def generate(
                 priority: 1) from the `generation_config.json` model file, if it exists; 2) from the model
                 configuration. Please note that unspecified parameters will inherit [`~generation.GenerationConfig`]'s
                 default values, whose documentation should be checked to parameterize generation.
+            pad_to_max_mel_tokens (`int`, *optional*):
+                Pads generated speech_ids to the specified value. This is to implement the same logic from the official
+                repo, link: https://github.com/neonbjb/tortoise-tts/blob/80f89987a5abda5e2b082618cd74f9c7411141dc/tortoise/api.py#L430
+                and to make sure the logits are same.
             output_hidden_states (`bool`, *optional*):
                 Whether or not to return the hidden states of decoder model, text encoder and speech encoder models.
 
         Returns:
             `ClvpOutput` or tuple: A `ClvpOutput` (if `return_dict_in_generate=True` or when
             `config.return_dict_in_generate=True`) or a tuple.
         """
+
+        # If the input sequences are larger than (self.config.decoder_config.max_text_tokens - 3) then raise error, because we add 1 bos tokens and
+        # 2 eos tokens to the input_ids in ClvpConditioningEncoder.
```

Comment:
```suggestion
        # If the input sequences are larger than (self.config.decoder_config.max_text_tokens - 3) then raise error, because we need to add 3 tokens ( 1 bos tokens and
        # 2 eos tokens) to the input_ids in ClvpConditioningEncoder to properly sample
```